lib/go/thrift/simple_server.go

```diff
@@ -356,6 +361,7 @@ func (p *TSimpleServer) processRequests(client TTransport) (err error) {
 		ok, err := processor.Process(ctx, inputProtocol, outputProtocol)
 		if errors.Is(err, ErrAbandonRequest) {
 			err := client.Close()
+			client = nil
```

Comment:
this does not work as you expected.

here `client` is the local arg of `processRequests` function, setting it to `nil` only affects that copy of `client`. it does not affect the `client` you are checking on line 214, which is at the caller of `processRequests`.

the whole nil checking regarding client is also unnecessary, if `client` is already closed, closing it again will only give you an already closed error, which you ignored anyways (on line 215).

Reply:
Makes sense. Then is it good to have ```defer client.Close``` in processRequests()?

Reply:
I would prefer to defer it in `innerAccept` instead.